lib/cloudregister/registerutils.py

```diff
@@ -2394,3 +2437,90 @@ def __set_registries_conf_docker(private_registry_fqdn):
         return write_registries_conf(
             docker_cfg_json, DOCKER_CONFIG_PATH, 'docker'
         )
+
+
+# ----------------------------------------------------------------------------
+def __set_registry_fqdn_suma(private_registry_fqdn):
+    """Set the registry FQDN for SUMa SUMA_REGISTRY_CONF_PATH file."""
+    suma_yaml_content = {}
+    os.makedirs(os.path.dirname(SUMA_REGISTRY_CONF_PATH), exist_ok=True)
+    suma_yaml_content, failed = get_suma_registry_content()
+    if failed:
+        return False
+
+    if suma_yaml_content.get('registry', {}) != private_registry_fqdn:
+        # registry value does not match the update server registry FQDN
+        # or there is no registry entry
+        # set the content
+        suma_yaml_content.update({'registry': private_registry_fqdn})
+        return __write_suma_conf(suma_yaml_content)
+
+    # the registry value inside the file has the update server registry FQDN
+    return True
+
+
+# ----------------------------------------------------------------------------
+def __write_suma_conf(updated_content):
+    """Update the SUMA SUMA_REGISTRY_CONF_PATH file with the new content."""
+    try:
+        with open(SUMA_REGISTRY_CONF_PATH, 'w') as suma_config:
+            yaml.dump(updated_content, suma_config, default_flow_style=False)
+        logging.info('%s updated' % SUMA_REGISTRY_CONF_PATH)
+        return True
+    except IOError as error:
+        logging.info(str(error))
+        action = 'open'
+    except yaml.YAMLError:
+        action = 'parse'
+
+    message = 'Could not %s %s' % (action, SUMA_REGISTRY_CONF_PATH)
+    logging.info(message)
+
+
+# ----------------------------------------------------------------------------
+def __unset_registry_fqdn_suma():
+    """Set the registry FQDN for SUMa SUMA_REGISTRY_CONF_PATH file."""
+    if not os.path.exists(SUMA_REGISTRY_CONF_PATH):
+        return True
+
+    suma_yaml_content = {}
+    suma_yaml_content, failed = get_suma_registry_content()
+    if failed:
+        return False
+
+    if not suma_yaml_content or suma_yaml_content == {'registry': ''}:
+        # empty content or no other content than registry key and empty value
+        os.unlink(SUMA_REGISTRY_CONF_PATH)
```

Comment:
Would removing that file harm SUMA ?

Reply:
When I asked that I ws told if the file is not present SUMA would use the builtin defaults, however, the image
should contain that file and it should not get removed

Will edit the code to reflect that

Reply:
if the file should be removed and the file should be baked in with the image + containing the registry FQDN
maybe should only add it if not present and let it be, meaning not removing that entry when deregistering 

Thoughts ?

Reply:
I have removed the SUMA file cleanup, we set the value if not present as the value should be present and `cloud-regionsrv-client` does not touch that file once value is set

Reply:
perfect, so that's clear then. Thanks

Reply:
Another point that we need to clarify. When we write that file, what needs to be done that SUMA reads it ? Maybe we need to send a signal or restart a service, or at least document the needed steps regarding suma after the registration. @jesusbv you mentioned a meeting with the SUMA people, can you ask that question there ? Thanks